### PR TITLE
New registry viewer build process

### PR DESCRIPTION
**Pull request by michael-valdron**
**What does this PR do / why we need it**:

Provides the following changes:
- Additional environment variables and build arguments to image build `Dockerfile`
- Adds additional shell scripts for building and publishing the registry viewer
- Add a push image workflow for building the registry viewer image, publishing the registry viewer image, and sending a dispatch to devfile/registry to trigger staging deployment workflows

**Which issue(s) this PR fixes**:

Fixes #?

part of devfile/api#747

**PR acceptance criteria**:

- [ ] Unit Tests
- [ ] E2E Tests
- [ ] Documentation

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by michael-valdron** [comment]
This PR is Blocked, the following repository secrets needs adding:

- `NEXT_PUBLIC_ANALYTICS_WRITE_KEY`: Analytics key for the registry viewer
- `QUAY_USERNAME`: For logging into the account for publishing to [quay.io/devfile](https://quay.io/organization/devfile)
- `QUAY_PASSWORD`: For logging into the account for publishing to [quay.io/devfile](https://quay.io/organization/devfile)
- `REPO_ACCESS_TOKEN`: For dispatching to [devfile/registry](https://github.com/devfile/registry)

cc: @johnmcollier @schultzp2020

**Comment by michael-valdron** [comment]
PR unblocked.

File: .github/workflows/pushimage-next.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# limitations under the License.
+
+name: Next Dockerimage
+
+on:
+  push:
+    branches: [main]
+  repository_dispatch:
+    types: [build]
+
```
Comment (schultzp2020) [inline .github/workflows/pushimage-next.yml:22]: Add
```yaml
concurrency:
  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
  cancel-in-progress: true
```

File: .github/workflows/pushimage-next.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+jobs:
+  registryViewerBuild:
```
Comment (schultzp2020) [inline .github/workflows/pushimage-next.yml:24]: jobs names should follow kebab case and should have a display name
i.e.
```yaml
jobs:
  registry-viewer-build:
    name: Registry Viewer Build
```

File: .github/workflows/pushimage-next.yml (hunk 1 above)
Comment (michael-valdron) [reply .github/workflows/pushimage-next.yml:22]: Done.

File: .github/workflows/pushimage-next.yml (hunk 2 above)
Comment (michael-valdron) [reply .github/workflows/pushimage-next.yml]: Changes applied.

File: .github/workflows/pushimage-next.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  registry-viewer-build:
+    name: Registry Viewer Build
+    runs-on: ubuntu-latest
+    env:
+      NEXT_PUBLIC_ANALYTICS_WRITE_KEY: ${{ secrets.NEXT_PUBLIC_ANALYTICS_WRITE_KEY }}
```
Comment (schultzp2020) [inline .github/workflows/pushimage-next.yml:32]: The write key is not a secret

File: .github/workflows/pushimage-next.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      - name: Build the registry viewer image
+        run: bash ./scripts/build_viewer.sh
+      - name: Push the registry viewer image
+        run: bash ./scripts/push.sh registry-viewer:latest quay.io/devfile/registry-viewer:next
+  dispatch:
+    needs: registryViewerBuild
```
Comment (schultzp2020) [inline .github/workflows/pushimage-next.yml:47]: This needs to be updated

**Review by schultzp2020 — APPROVED** [review]

**Pull request merged**